account/src/main/java/com/hms/lib/commonmobileservices/account/google/GoogleAccountServiceImpl.kt

```diff
@@ -97,4 +99,31 @@ internal class GoogleAccountServiceImpl(private val context: Context, signInPara
         }
     }
 
+    override fun startSmsRetriver(context: Context): Work<Unit> {
+        val worker: Work<Unit> = Work()
+
+        mGoogleSmsRetriver.startSmsRetriever().addOnSuccessListener { worker.onSuccess(Unit) }
+            .addOnFailureListener { worker.onFailure(it) }
+            .addOnCanceledListener { worker.onCanceled() }
+        return worker
+    }
+
+    override fun startConsent(activity: Activity, phonenumber: String): Work<Unit> {
+        val worker: Work<Unit> = Work()
+
+        mGoogleSmsRetriver.startSmsUserConsent(phonenumber)
+            .addOnSuccessListener { worker.onSuccess(Unit) }
+            .addOnFailureListener { worker.onFailure(it) }
+            .addOnCanceledListener { worker.onCanceled() }
+        return worker
+    }
+
+    override fun getSignAccountId(): SignInUser? {
+        return (mGoogleSignInResult as GoogleSignInResult).signInAccount?.let { mapper.map(it) }
```

Comment:
Did you test this method? mGoogleSignInResult has never been initialized it will be null.